src/dotnet/APIView/APIViewWeb/Pages/Assemblies/Conversation.cshtml.cs

```diff
@@ -86,5 +96,90 @@ public async Task<IActionResult> OnGetAsync(string id)
         }
+
+        private Dictionary<UsageSampleRevisionModel, List<CommentThreadModel>> ParseUsageSampleThreads(IEnumerable<CommentThreadModel> threads)
+        {
+            var threadDict = new Dictionary<UsageSampleRevisionModel, List<CommentThreadModel>>();
+            
+            if (Sample == null)
+            {
+                return threadDict;
+            }
+
+            foreach (var thread in threads)
+            {
+                foreach (var comment in thread.Comments)
+                {
+                    if (comment.IsUsageSampleComment)
+                    {
+                        var sampleRevision = Sample.Revisions.Where(e => e.FileId.Equals(comment.ElementId.Split("-").First())).First();
+                        if (sampleRevision.RevisionIsDeleted)
+                        {
+                            continue;
+                        }
+
+                        if(!threadDict.ContainsKey(sampleRevision))
+                        {
+                            threadDict.Add(sampleRevision, new List<CommentThreadModel>());
+                        }
+                        threadDict[sampleRevision].Add(thread);
+
+                    }
+                }
+            }
+
+            return threadDict;
+        }
+
+        private async Task<IEnumerable<List<string>>> getUsageSampleLines()
+        {
+            List<List<string>> lines = new List<List<string>>();
+
+            foreach (var revision in Sample.Revisions.ToArray().Reverse())
+            {
+                if (revision.RevisionIsDeleted)
+                {
+                    lines.Add(new List<string>());
+                    if (revision.RevisionNumber == 1)
+                    {
+                        break;
+                    }
+                    continue;
+                }
+
+                string rawContent = await _samplesManager.GetUsageSampleContentAsync(revision.FileId);
+                var sampleLines = new List<string>();
+
+                rawContent = rawContent.Trim();
+                string[] content = rawContent.Split("\n");
+
+                for (int i = 0; i < content.Length; i++)
+                {
+                    string lineContent = content[i];
+                    if (lineContent.StartsWith("<div class=\"code") || lineContent.StartsWith("<div style=") || lineContent.StartsWith("<div class=\"lang"))
+                    {
+                        lineContent = "";
+                    }
+
+                    lineContent = lineContent.Replace("<pre>", "").Replace("</pre>", "").Replace("</div>", "").Replace("\n", "");
+                    if (lineContent.Trim() == "")
+                    {
+                        continue;
+                    }
+
+                    sampleLines.Add(lineContent);
+                }
+
+                lines.Add(sampleLines);
+
+                if (revision.RevisionNumber == 1)
```

Comment:
Why are we breaking at revision number 1?

Reply:
I'm not quite sure. I believe this was a workaround for deleted revisions used before I changed the system, then copy and pasted over without proper checking. Removed both this and the deleted check further up